### PR TITLE
Add a "terminal" callback to TastyInspector

### PR DESCRIPTION
**Pull request by abgruszecki**
@nicolasstucki I think we should allow defining a "final" callback in TastyInspector that runs after all compilation units are processed. The reason is very simple - this way, users can encode arbitrary operations on the full list of compilation units.

I've found a bunch of uses for that in the doctool and I suspect that external users will quickly need this as well.

### CODE REVIEW

File: tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala [hunk 1]
```diff
@@ -89,6 +93,19 @@ trait TastyInspector:
...
     end TastyInspectorPhase
 
+    class TastyInspectorFinishPhase extends Phase:
+
+      override def phaseName: String = "tastyInspectorFinish"
+
+      override def run(implicit ctx: Context): Unit =
+        if !alreadyRan then
+          val qctx = QuoteContextImpl()
+          self.finishProcessingCompilationUnits(using qctx)
```
Comment (romanowski) [inline tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala:103]: What if `finishProcessingCompilationUnits` throws exception? Shound't we use 
`try self.finishProcessingCompilationUnits(using qctx) finally alreadyRan = true` here?

Comment (abgruszecki) [reply tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala:103]: I think that this is already handled in the code that orchestrates Phase runs, Tasty Inspector will continue even if `processCompilationUnit` throws an exception and there's no special code for doing that above.

Comment (romanowski) [reply tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala:103]: My point was not to run `finishProcessingCompilationUnits` twice in such case (there was only `finally` there without `catch`)

File: tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala [hunk 2]
```diff
@@ -89,6 +93,21 @@ trait TastyInspector:
 
     end TastyInspectorPhase
 
+    class TastyInspectorFinishPhase extends Phase:
+
+      override def phaseName: String = "tastyInspectorFinish"
+
+      override def run(implicit ctx: Context): Unit =
```
Comment (nicolasstucki) [inline tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala:100]: We should override `runOn` and not  have `alreadyRan`.

File: tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala [hunk 3]
```diff
@@ -20,6 +20,9 @@ trait TastyInspector:
   /** Process a TASTy file using TASTy reflect */
   protected def processCompilationUnit(using QuoteContext)(root: qctx.reflect.Tree): Unit
 
+  /** Called after all compilation units are processed */
+  protected def finishProcessingCompilationUnits(using QuoteContext): Unit = ()
```
Comment (nicolasstucki) [inline tasty-inspector/src/scala/tasty/inspector/TastyInspector.scala:24]:
```suggestion
  protected def postProcess(using QuoteContext): Unit = ()
```

**Review by nicolasstucki — CHANGES_REQUESTED** [review]

**Pull request merged**